src/main/java/com/redhat/sast/api/service/UrlValidationService.java

```diff
@@ -0,0 +1,55 @@
+package com.redhat.sast.api.service;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import lombok.extern.slf4j.Slf4j;
+
+@ApplicationScoped
+@Slf4j
+public class UrlValidationService {
+
+    private final HttpClient httpClient;
+
+    public UrlValidationService() {
+        this.httpClient =
+                HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(5)).build();
+    }
+
+    public boolean isUrlAccessible(String url) {
+        if (url == null || url.trim().isEmpty()) {
+            LOGGER.debug("URL is null or empty, returning false");
+            return false;
+        }
+
+        try {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(url))
+                    .method("HEAD", HttpRequest.BodyPublishers.noBody())
+                    .timeout(Duration.ofSeconds(10))
```

Comment:
The timeout of an httpClient is a very crucial configuration. It often causes production errors and requires a lot of time to figure it out. I can see, the default value is set to 5 seconds, but here it is increased to 10 seconds. Why is there a difference? It would be better to define a configurable value through application.properties. Also, make it clear that we are now using 2 different HTTP clients, and it can cause confusion for other developers when setting such config values.

Reply:
[94f6673](https://github.com/RHEcosystemAppEng/sast-ai-orchestrator/pull/50/commits/94f66735d23898c03fc485f54ae3fb0c6781e8e1)